driver-core/src/test/java/com/datastax/driver/core/TypeCodecJodaIntegrationTest.java

```diff
@@ -0,0 +1,166 @@
+     * <p>
+     * Validates that a <code>date</code> column can be mapped to a {@link org.joda.time.LocalDate} by using
+     * {@link com.datastax.driver.core.JodaCodecs.LocalDateCodec}.
+     * </p>
+     *
+     * @test_category data_types:serialization
+     * @expected_result properly maps.
+     * @jira_ticket JAVA-721
+     * @since 2.2.0
+     */
+    @Test(groups="short")
+    public void should_map_date_to_localdate() {
+        org.joda.time.LocalDate date = new org.joda.time.LocalDate(2015, 1, 1);
+        LocalDate lDate = LocalDate.fromYearMonthDay(2015, 1, 1);
+
+        session.execute("insert into foo (c1, cd) values (?, ?)", "should_map_date_to_localdate", date);
+
+        ResultSet result = session.execute("select cd from foo where c1=?", "should_map_date_to_localdate");
+
+        assertThat(result.getAvailableWithoutFetching()).isEqualTo(1);
+
+        Row row = result.one();
+        assertThat(row.getObject("cd", org.joda.time.LocalDate.class)).isEqualTo(date);
+        assertThat(row.getDate("cd")).isEqualTo(lDate);
+    }
+
+
+    /**
+     * <p>
+     * Validates that a <code>timestamp</code> column can be mapped to a {@link DateTime} by using
+     * {@link com.datastax.driver.core.JodaCodecs.DateTimeCodec}.
+     * </p>
+     *
+     * @test_category data_types:serialization
+     * @expected_result properly maps.
+     * @jira_ticket JAVA-721
+     * @since 2.2.0
+     */
+    @Test(groups="short")
+    public void should_map_timestamp_to_datetime() {
+        DateTime dateTime = DateTime.parse("2010-06-30T01:20+05:00");
+        Date date = dateTime.toDate();
+        DateTimeZone timeZone = dateTime.getZone();
+
+        session.execute("insert into foo (c1, ctimestamp) values (?, ?)", "should_map_timestamp_to_datetime", dateTime);
+
+        ResultSet result = session.execute("select ctimestamp from foo where c1=?", "should_map_timestamp_to_datetime");
+
+        assertThat(result.getAvailableWithoutFetching()).isEqualTo(1);
+
+        Row row = result.one();
+        // Since timezone is not preserved in timestamp type, set it.
+        DateTime rDateTime = row.getObject("ctimestamp", DateTime.class).withZone(timeZone);
+        assertThat(rDateTime).isEqualTo(dateTime);
+        assertThat(row.getTimestamp("ctimestamp")).isEqualTo(date);
+    }
+
+
+    /**
+     * <p>
+     * Validates that a <code>tuple&lt;timestamp,text&gt;</code> column can be mapped to a {@link DateTime} by using
+     * {@link com.datastax.driver.core.JodaCodecs.TimeZonePreservingDateTimeCodec}.
+     * </p>
+     *
+     * @test_category data_types:serialization
+     * @expected_result properly maps.
+     * @jira_ticket JAVA-721
+     * @since 2.2.0
+     */
+    @Test(groups="short")
+    public void should_map_tuple_to_datetime() {
+        // Register codec that maps DateTime <-> tuple<timestamp,varchar>
+        cluster.getConfiguration().getCodecRegistry().register(new JodaCodecs.TimeZonePreservingDateTimeCodec(cluster));
```

Comment:
Although, I changed the constructor to take a TupleCodec as its only argument; at least I think that the instantiation of TimeZonePreservingDateTimeCodec is a bit less convoluted this way.